xdesign/plot.py

```diff
@@ -367,26 +377,54 @@ def discrete_phantom(phantom, size, ratio=9, uniform=True,
     image = 0
 
     if phantom.geometry is not None and phantom.material is not None \
-       and hasattr(phantom.material, prop):
+       and np.alltrue([hasattr(phantom.material, temp) for temp in prop]):
 
-        psize = 1.0 / size
+        # select whether use a fixed pixel size (at 1) or fixed object size (1cm)
+        if fix_psize:
+            psize = 1
+        else:
+            psize = 1.0 / size
 
         # Rasterize all geometry in the phantom.
         pmin, patch = discrete_geometry(phantom.geometry, psize, ratio)
 
         # Get the property value
-        value = getattr(phantom.material, prop)(DEFAULT_ENERGY)
-
-        # Make a grid to put store all of the discrete geometries
-        image = np.zeros([size] * phantom.geometry.dim, dtype=float)
+        prop = np.atleast_1d(prop)
+        n_prop = prop.size
+        ret = [None] * n_prop
+        for i, i_prop in enumerate(prop):
+            try:
+                value = getattr(phantom.material, i_prop)(energy)
+            except:
+                value = getattr(phantom.material, i_prop)
```

Comment:
First, we shouldn't have a bare exception. We should always name the exception we intend to catch. I think in this case we are looking for `TypeError` because the energy argument will be rejected if the attribute isn't callable.

Related: what if we just change all material properties to this signature:
```python
def some_material_property(self, **kwargs):
    energy = kwargs['energy']
```
Then an error is only raised if you're trying to get a property without enough info, but if you have extra info, it doesn't matter.